runtime/actors/agents.go

```diff
@@ -34,16 +35,14 @@ type AgentInfo struct {
 
 type AgentStatus = string
 
-// TODO: validate these statuses are needed and used correctly
 const (
-	AgentStatusUninitialized AgentStatus = "uninitialized"
-	AgentStatusError         AgentStatus = "error"
-	AgentStatusStarting      AgentStatus = "starting"
-	AgentStatusStarted       AgentStatus = "started"
-	AgentStatusStopping      AgentStatus = "stopping"
-	AgentStatusStopped       AgentStatus = "stopped"
-	AgentStatusSuspended     AgentStatus = "suspended"
-	AgentStatusTerminated    AgentStatus = "terminated"
+	AgentStatusStarting    AgentStatus = "starting"
+	AgentStatusRunning     AgentStatus = "running"
+	AgentStatusSuspending  AgentStatus = "suspending"
+	AgentStatusSuspended   AgentStatus = "suspended"
+	AgentStatusRestoring   AgentStatus = "restoring"
+	AgentStatusTerminating AgentStatus = "terminating"
+	AgentStatusTerminated  AgentStatus = "terminated"
```

Comment:
nit: it seems we're not using these outside the package, should we make it lowercase to not export it?

Reply:
Done.  I updated all object names in the `actors` package so that only the necessary items are exported.

This had one interesting side effect, when changing the field names of the `agentInfo` struct to lower case, it then caused an exception in some of the language support code, because reflection was used to export those fields.  There was similar code in three places, so I extracted to a common function, added a utility function, and then updated the reflection code to account for non-exported fields.